### PR TITLE
Create Contacts

### PR DESCRIPTION
**Pull request by nicolas-brousse**
- [x] Add tests
- [x] Make Contact and ContactRole index columns sortables
- [x] Implement assignments, create CRUD and frame like for associations
- [x] I18n
- [x] List assignments on Contact show
- [x] In contact index regroup first_name and last_name

### CODE REVIEW

**Comment by B-Rass** [comment]
Ok I think I'm done with it. 

I thought it may be nice to always show contact assignment by default

**Comment by nicolas-brousse** [comment]
> I thought it may be nice to always show contact assignment by default

Make sense, we have to move assignments index table in a partial and use it directly in the site show view. I'll look at it when I rework on this.

File: app/controllers/contact_roles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        format.html { render :edit }
+        format.json { render json: @contact_role.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /contact_roles/1
+  # DELETE /contact_roles/1.json
+  def destroy
+    if @contact_role.destroy
```
Comment (B-Rass) [inline app/controllers/contact_roles_controller.rb:58]: I had a Rails error when I try to delete Role with an assignment.

Can't we just show a flash error message?

File: config/locales/fr.yml [hunk 2]
```diff
@@ -653,7 +657,61 @@ fr:
...
+      flashes:
+        updated: Le role a bien été mis à jour.
+    destroy:
+      flashes:
+        destroyed: Le role a bien été supprimé.
+  contact_assignments:
+    index:
+      new_contact_assignment: Ajouter une assignation
+      title: Assignations de Contacts
+      delete_confirmation: Cette assignation ne pourra pas être supprimé tant qu'il est associé à au moins un Site. Voulez-vous continuer ?
```
Comment (B-Rass) [inline config/locales/fr.yml:701]: I can delete with an assignation, so we should change the message here

File: test/fixtures/contact_assignments.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# one:
+#   site: one
+#   contact: one
+#   contact_role: one
+
+# two:
+#   site: two
+#   contact: two
+#   contact_role: two
```
Comment (B-Rass) [inline test/fixtures/contact_assignments.yml:11]: Why is it commented?

File: config/locales/fr.yml [hunk 4]
```diff
@@ -653,7 +657,61 @@ fr:
         updated: La baie a bien été mise à jour.
     destroy:
       flashes:
-        destroyed: La baie bien été supprimé.
+        destroyed: La baie a bien étée supprimé.
+  contacts:
+    index:
+      new_contact: Ajouter un contact
+      title: Contacts
+      delete_confirmation: Ce contact ne pourra pas être supprimé tant qu'il est associé à au moins un Site. Voulez-vous continuer ?
```
[the remark this answers is not in the document]
Comment (nicolas-brousse) [reply config/locales/fr.yml]: I don't think we should disallow user to remove a contact with assignation. But change the message to be coherent yes.

File: app/controllers/contact_roles_controller.rb (hunk 1 above)
Comment (nicolas-brousse) [reply app/controllers/contact_roles_controller.rb:58]: I think there is a config missing in the model.

File: test/fixtures/contact_assignments.yml (hunk 3 above)
Comment (nicolas-brousse) [reply test/fixtures/contact_assignments.yml:11]: Not used and doesn't work if uncommented

Comment (B-Rass) [reply test/fixtures/contact_assignments.yml:11]: ok

File: app/controllers/contact_roles_controller.rb (hunk 1 above)
Comment (nicolas-brousse) [reply app/controllers/contact_roles_controller.rb:58]: fixed

File: config/locales/fr.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (nicolas-brousse) [reply config/locales/fr.yml]: fixed

File: config/locales/fr.yml (hunk 2 above)
Comment (nicolas-brousse) [reply config/locales/fr.yml]: fixed

**Review by B-Rass — APPROVED** [review]

**Pull request merged**